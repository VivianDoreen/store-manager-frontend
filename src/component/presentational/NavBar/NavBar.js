import React, { Component } from 'react';
import { connect } from 'react-redux';
import NavLink from 'react-router-dom/NavLink';

export class NavBar extends Component {
    state={
        isAuthenticated:false,
    }
    componentWillMount(){
        const token = localStorage.getItem('token')
        if(token){
            this.setState({
                isAuthenticated:true
            })
        }

    }
    render() {
        return (
            <div id= "navbar">
                <h1 id="headerSM"> StoreManager</h1>
                <div id="navDiv">
                    <span>
                        <NavLink to='/' className="navLink">Home</NavLink>
                    </span>
                    {
                    this.state.isAuthenticated && (this.props.signin.data.role == 'admin') ?
                    <span>
                        <NavLink to='/signup' id='avatar' className="navLink">
                            Register
                        </NavLink>
                        <NavLink to='/login' className="navLink" onClick={() => localStorage.removeItem('token')}>
                                Logout
                        </NavLink>
                    </span>:
                    this.state.isAuthenticated && (this.props.signin.data.role == 'attendant') ?
                    <span>
                        <NavLink to='/' id='avatar' className="navLink">
                            Sales
                        </NavLink>
                        <NavLink to='/login' className="navLink" onClick={() => localStorage.removeItem('token')}>
                                Logout
                        </NavLink>
                    </span>:
                    <span>
                        <a href="/login" id="loginId">Login</a>
                    </span>
                    
                }
                </div>
            </div> 
        );
    }
}

export const MapStateToProps = state => {
    return state.navBarReducer 
};

export default connect(MapStateToProps)(NavBar);

