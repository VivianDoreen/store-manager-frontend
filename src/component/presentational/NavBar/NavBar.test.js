import React, { Component } from 'react';
import {NavBar} from '../NavBar/NavBar';
import {shallow} from 'enzyme';

describe('Navabr', () =>{
    let wrapper;
    wrapper = shallow(<NavBar />)

    it('should render without crashing', () => {

        expect(wrapper).toMatchSnapshot();
    
      });

})