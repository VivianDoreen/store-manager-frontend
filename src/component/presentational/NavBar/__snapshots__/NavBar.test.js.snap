// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navabr should render without crashing 1`] = `ShallowWrapper {}`;
